### PR TITLE
Add UserGuide.md documentation for Topic Search

### LINKED ISSUE #27: User Documentation: Post Searching
From the write-up:

* Create a new file `UserGuide.md` in the root directory of your repository
* In this file, provide a detailed outline of how to use and user test your new feature(s)
* You should also provide a link/description of where your added automated tests can be found, along with a description of what is being tested and why you believe the tests are sufficient for covering the changes that you have made.

This issue tracks the documentation entry for post searching.

### PR DESCRIPTION
**Pull request by brycez1**
Added user guide documentation for Topic Search.

- Demo Video
- Testing Instructions
- Discussion of Automated Tests

Added to existing UserGuide.md file in root directory of repo.

Would resolve #27.

### CODE REVIEW

File: UserGuide.md [hunk 1]
```diff
@@ -29,4 +29,31 @@ If the `iss8` branch is merged, the tests can be found in `tests/topics.js`. We
...
+5. Ensure that the search toggle is set to "titles", "posts", or "titles and posts".
+6. Locate the "topic name" search field and then add a filter for the desired topic.
+7. You can now fill in all of the other fields as desired and will be able to filter for topic names.
+
+### Automated Tests
+The tests for this feature can be found in `tests/search.js`. 
+- Tests that the search finds the correct post if the correct topic name is placed in the filter.
+- Returns nothing if no posts match the filter description.
+
+I think these tests are sufficient because they ensure that the filter functions as necessary by finding posts that match the query while also working correctly if nothing is found. For displaying posts with the correct topic name, is successfully filters out the other post which matches the other query descriptions.
```
Comment (tkroening) [inline UserGuide.md:57]: Typo here?

> is successfully filters out the other post

**Review by tkroening — DISMISSED** [review]
Looks good! I think this hits all the bullets.

File: UserGuide.md (hunk 1 above)
Comment (brycez1) [reply UserGuide.md:57]: Oh woops, I'll fix that.

**Review by tkroening — APPROVED** [review]
Alright, I think this is good to go!

**Review by Chen3018 — APPROVED** [review]
LGTM. The guide is detailed and accurate

**Pull request merged**